src/programStage/FieldProgramStageWithLoadingStates.js

```diff
@@ -15,10 +27,20 @@ export const FieldProgramStageWithAutoLoad = ({ required, programId }) => {
         if (programId) refetch({ programId })
     }, [programId])
 
+    useEffect(() => {
+        const initialValue =
+            parserType === EVENT_REGISTRATION_PARSER.value
+                ? { id: data?.programStages?.programStages[0]?.id }
+                : undefined
```

Comment:
Maybe you could split this up into several statements to improve the readabliity.

Reply:
Done